lib/chef/resource/package.rb

```diff
@@ -31,14 +31,13 @@ class Package < Chef::Resource
 
       def initialize(name, run_context=nil)
         super
-        @candidate_version = nil
-        @options = nil
-        @package_name = name
-        @response_file = nil
-        @response_file_variables = Hash.new
-        @source = nil
-        @version = nil
-        @timeout = nil
+        options                 nil
+        package_name            name
+        response_file           nil
+        response_file_variables Hash.new
+        source                  nil
+        version                 nil
+        timeout                 nil
```

Comment:
Yeah it'll also magically turn into a setter in 13.0 which will DTRT, and if it blows up because the setter doesn't support setting a nil then that's also a clear bug to fix since any property which starts out as nil needs to be nillable so that the user can reset it (so needs NilClass added or whatever).